src/main/java/uk/gov/justice/digital/delius/service/ReferralService.java

```diff
@@ -12,24 +12,43 @@
 import uk.gov.justice.digital.delius.data.api.deliusapi.NewNsiManager;
 
 import java.util.Collections;
+import java.util.Map;
 import java.util.Optional;
 
 import static java.util.stream.Collectors.toList;
 
 @Service
 public class ReferralService {
-    public static final String UNALLOCATED_STAFF_SUFFIX = "UATU";
-    public static final String UNALLOCATED_TEAM_SUFFIX = "UAT";
+
+    private final String providerCode;
+    private final String staffCode;
+    private final String teamCode;
+    private final String nsiStatus;
+    private final Map<String, String> referralTypeToNsiTypeMapping;
+
     private final DeliusApiClient deliusApiClient;
 
     private final NsiService nsiService;
 
     private final OffenderService offenderService;
 
-    public ReferralService(final DeliusApiClient deliusApiClient, final NsiService nsiService, final OffenderService offenderService) {
+    public ReferralService(final DeliusApiClient deliusApiClient,
+                           final NsiService nsiService,
+                           final OffenderService offenderService,
+                           @Value("${new-nsi.referral.provider-code}") final String providerCode,
```

Comment:
You might want to group these into a config object using @ConfigurationProperties(prefix = "new-nsi")

Reply:
Will do